### ISSUE TITLE
Any wsl command is met with access denied after installing ubuntu 24

### ISSUE THREAD

**Issue by jsandr** [microsoft/WSL#11560]
### Windows Version

Microsoft Windows [version 10.0.22631.3527]

### WSL Version

Access denied

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Access denied

### Distro Version

Ubuntu 24.04 LTS

### Other Software

Wsl Ubuntu 20.04.6 LTS
Windows Terminal Version : 1.19.11213.0

I had wsl2 with ubuntu 20 for a while, no issue there. I wanted to switch to the latest LTS and added ubuntu 24 through the store. Using Terminal it worked fine. I did not want to unregister the previous ubuntu version until I was sure I would still have everything working fine. Booting my computer this morning, every wsl command is met with '_access denied_'.

### Repro Steps

Install Ubuntu 24 through the store, reboot the computer.
1. Open a terminal and run > wsl

2. Try to read wsl installed distros: > wsl -l

### Expected Behavior

1. Wsl default distro is launched.
2. Powershell returns the list of installed linux distributions

### Actual Behavior

1.
```
PS C:\Users\jsandrin> wsl
Accès refusé.
```
2.
```
PS C:\Users\jsandrin> wsl -l
Accès refusé.
```

**Comment by jsandr** [comment]
Similar issues seems to all have been closed without author answers.

Launching Ubuntu from the store does not work either.

I just found out that I can run the wsl program but not the 'ubuntu' one.
Inside wsl I could check my version:
```
jsandr@SST-JSA:~$ uname -a
Linux SST-JSA 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
As shown in this screenshot I was able to run wsl but not ubuntu
![ubu](https://github.com/microsoft/WSL/assets/61015242/2e5c9f6d-3f22-4aeb-b7ed-c24c9fcfdbda)

**Comment by OneBlue** [comment]
Thank you @jsandr. 

Looking at the logs, this looks like the MSIX installation might be broken, but I wonder if you can call wsl.exe directly.

Does calling: C:\Program Files\WSL\wsl.exe work ?

**Comment by jsandr** [comment]
Hi @OneBlue, yes _wsl.exe_ does work.

**Comment by OneBlue** [comment]
> Hi @OneBlue, yes _wsl.exe_ does work.

Ok then this is indeed an MSIX issue. 

Does re-installing the latest [package](https://github.com/microsoft/WSL/releases/download/2.2.4/wsl.2.2.4.0.x64.msi) solve the issue ?

**Comment by jsandr** [comment]
Hi,
Re-installing throws this: 
![install_ubu](https://github.com/microsoft/WSL/assets/61015242/01e8d7f6-9498-4cf6-8751-17e1e5c09579)

As any wsl commands are refused I could not un-register Ubuntu 24, so I went to 'add or remove apps' and uninstalled both Ubuntu 20 and 24. Removed all startup calls to wsl or terminal. Rebooted then tried install via the .msi
After it failed I went to the store, installed Ubuntu 24 again (it worked ?!). Now I'm back with an Ubuntu 24 app but still unable to launch it directly.
Only noticeable difference is that now when trying to launch a wsl command via powershell this is the answer I have:
![wsl_shut](https://github.com/microsoft/WSL/assets/61015242/9afa5bda-10cd-45c8-88be-46f5d76952ef)

**Comment by jsandr** [comment]
Hi, I just found out that my home is not at ~ but is at ü
![home](https://github.com/microsoft/WSL/assets/61015242/a037fc1f-0edd-4e9b-85b1-1b5366218e8a)

**Comment by jsandr** [comment]
Anyone else with a better idea than re-installing in loops until it starts working again ?

**Comment by withinboredom** [comment]
No, but I'm also having massive issues with ubuntu 24. After about 10 minutes, systemd entirely locks up (took forever just to get that much information) and it seems the last windows update did it in. Super frustrating.

**Comment by Blaksey** [comment]
> > Hi @OneBlue, yes _wsl.exe_ does work.
> 
> Ok then this is indeed an MSIX issue.
> 
> Does re-installing the latest [package](https://github.com/microsoft/WSL/releases/download/2.2.4/wsl.2.2.4.0.x64.msi) solve the issue ?

I'm having similar issues. Any wsl command is met with "Access is denied". If I run it from the Program Files it does seem fine but applying updates doesn't seem to resolve.
![image](https://github.com/microsoft/WSL/assets/2694815/a7253d0e-fb35-4a35-a29d-42743889ab5a)

 I've tried completely uninstalling and disabling the feature but no luck there either.

One strange thing I found is that if I go into the "Windows Subsystem for Linux" system component and then reset the component. The first command will run with WSL but after that it immediately goes back to "Access is denied" on subsequent commands. I'm using a relatively fresh Windows 11 install.

![image](https://github.com/microsoft/WSL/assets/2694815/090fe07b-3bc7-403d-9666-24184439afaa)

Example where I "reset" and then run two commands:
![image](https://github.com/microsoft/WSL/assets/2694815/9a56a17d-eae5-4c42-8ab0-42149dea497f)

**Comment by jsandr** [comment]
Hi, I installed windows updates KB5035942 and KB5037853 which enabled Ubuntu again.

> \> wsl -l
Distributions du Sous-système Windows pour Linux :
Ubuntu-24.04 (par défaut)
Ubuntu-20.04

Now I'm missing the GUI, apps like _meld_ or _gitg_ starts a process but remains hanging without launching a new window.

**Comment by withinboredom** [comment]
I wonder if this is related to a systemd bug that causes nearly the entire WSL instance to completely hang after running for about 10/15 minutes. IIRC, ubuntu 24 starts systemd by default (check in `/etc/wsl.conf`). I set it to `false` and things seem to be working much better.

**Comment by Shadow-Cipher** [comment]
> ### Windows Version
> Microsoft Windows [version 10.0.22631.3527]
> 
> ### WSL Version
> '_Access denied_' | ubutnu 24.04 LTS or Ubuntu 20.04.6 LTS
> 
> ### Are you using WSL 1 or WSL 2?
> * [x]  WSL 2
> * [ ]  WSL 1
> 
> ### Kernel Version
> Access denied
> 
> ### Distro Version
> Ubuntu 24.04 LTS
> 
> ### Other Software
> Wsl Ubuntu 20.04.6 LTS Windows Terminal Version : 1.19.11213.0 Powershell 7.4.2
> 
> I had wsl2 with ubuntu 20 for a while, no issue there. I wanted to switch to the latest LTS and added ubuntu 24 through the store. Using Terminal it worked fine. I did not want to unregister the previous ubuntu version until I was sure I would still have everything working fine. Booting my computer this morning, every wsl command is met with '_access denied_'.
> 
> Edit: Running as admin does no difference.
> 
> ### Repro Steps
> Install Ubuntu 24 through the store, reboot the computer.
> 
> 1. Open a terminal and run > wsl
> 2. Try to read wsl installed distros: > wsl -l
> 
> ### Expected Behavior
> 1. Wsl default distro is launched.
> 2. Powershell returns the list of installed linux distributions
> 
> ### Actual Behavior
> 
> ```
> PS C:\Users\jsandrin> wsl
> Accès refusé.
> ```
> 
> 
> ```
> PS C:\Users\jsandrin> wsl -l
> Accès refusé.
> ```
> 
> ### Diagnostic Logs
> [WslLogs-2024-05-06_10-09-13.zip](https://github.com/microsoft/WSL/files/15218242/WslLogs-2024-05-06_10-09-13.zip)



I migrated all of my WSL distros from the C: drive to a high capacity external drive about 2 years ago and since I did that, I have not had the same problems I was having with WSL. All my distros function pretty well now. There are some minor bugs using nano that are annoying but that's about all I've had issues with since. I have not seen a WSL related accessdenied error since then. I migrated them to reclaim some disk space on C:, but in the process it seems to have fixed the accessdenied errors I had been getting from time to time, so that was a nice bonus.
